cmd/storj/cmd/ls.go

```diff
@@ -32,49 +35,71 @@ func init() {
 func list(cmd *cobra.Command, args []string) error {
 	ctx := process.Ctx(cmd)
 
-	so, err := getStorjObjects(ctx, lsCfg)
+	identity, err := lsCfg.Load()
+	if err != nil {
+		return err
+	}
+
+	bs, err := lsCfg.GetBucketStore(ctx, identity)
 	if err != nil {
 		return err
 	}
 
 	if len(args) == 0 {
-		bi, err := so.ListBuckets(ctx)
-		if err != nil {
-			return err
+		startAfter := ""
+		var items []buckets.ListItem
+		for {
+			moreItems, more, err := bs.List(ctx, startAfter, "", 0)
+			if err != nil {
+				return err
+			}
+
+			items = append(items, moreItems...)
```

Comment:
instead of adding to a slice, you could just print the buckets out here?

Reply:
Done